crates/autopilot/src/run_loop.rs

```diff
@@ -0,0 +1,77 @@
+use anyhow::{anyhow, Context, Result};
+use chrono::Utc;
+use model::auction::{Auction, AuctionId};
+use shared::current_block::CurrentBlockStream;
+use std::time::Duration;
+
+use crate::{
+    database::Postgres, driver_api::Driver, driver_model::solve,
+    solvable_orders::SolvableOrdersCache,
+};
+
+const SOLVE_TIME_LIMIT: Duration = Duration::from_secs(15);
+
+pub struct RunLoop {
+    solvable_orders_cache: SolvableOrdersCache,
+    database: Postgres,
+    drivers: Vec<Driver>,
+    current_block: CurrentBlockStream,
+}
+
+impl RunLoop {
+    pub async fn run_forever(&self) -> ! {
+        loop {
+            match self.single_run().await {
+                Ok(()) => tracing::debug!("single run ok"),
+                Err(err) => tracing::error!(?err, "single run err"),
+            }
+            tokio::time::sleep(Duration::from_secs(1)).await;
+        }
+    }
+
+    async fn single_run(&self) -> Result<()> {
+        let auction = match self.solvable_orders_cache.current_auction() {
+            Some(auction) => auction,
+            None => {
+                tracing::debug!("no current auction");
```

Comment:
Isn't not having an auction is something we should `warn` about? Seems like this should only happen in pathological cases.

Reply:
This can happen on startup so it should be benign.